src/snowflake/cli/api/commands/snow_typer.py

```diff
@@ -91,12 +92,13 @@ def custom_command(command_callable):
             @wraps(command_callable)
             def command_callable_decorator(*args, **kw):
                 """Wrapper around command callable. This is what happens at "runtime"."""
-                self.pre_execute()
+                execution_id: str = uuid.uuid4().hex
+                self.pre_execute(execution_id)
                 try:
                     result = command_callable(*args, **kw)
```

Comment:
Can we also measure time of this execution so we can log it in telemetry data?

Reply:
I add start_time and end_time to a new event